main.go

```diff
@@ -26,8 +27,12 @@ func makeH2Request(
 		Timeout:   time.Duration(timeout) * time.Second,
 		Transport: tr,
 	}
-
-	req, err := http.NewRequest(method, url, requestBody)
+	b, err := io.ReadAll(requestBody)
```

Comment:
```suggestion
        // convert the buffer to a interface that supports `.Len()` so that Content-Length header is added
	b, err := io.ReadAll(requestBody)
```